project.mml

```diff
@@ -1621,6 +1621,7 @@ Layer:
               ) _
             ) AS features
           WHERE feature IS NOT NULL
+            AND (way_pixels < 768000 OR way_pixels IS NULL)
```

Comment:
Can you move this to inside the subselect, applying only to the polygon table and dropping the point case.

I'm not sure the planner will evaluate what amounts to `COALESCE(way_area/NULLIF(POW(!scale_denominator!*0.001*0.28,2),0), 0) < 768000` before st_pointonsurface.

Where we have a way_area condition, we have way_area > 1*!pixel_width!::real*!pixel_height!::real. iirc, this doesn't work work properly with different screen resolutions and scale factors, so I think this would work: `way_area < 768000 * POW(!scale_denominator!*0.001*0.28,2)`

Reply:
This is changed now - and also unified in amenity-points and text-poly-low-zoom.